build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryBaseClient.java

```diff
@@ -149,4 +157,44 @@ public boolean isRepoExist(String repo) throws IOException {
         }
         return true;
     }
+
+    public void reportUsage(UsageReporter usageReporter) throws VersionException, IOException {
+        // Check Artifactory supported version.
+        ArtifactoryVersion version = getArtifactoryVersion();
+        if (version.isNotFound()) {
+            throw new VersionException("Could not get Artifactory version.", VersionCompatibilityType.NOT_FOUND);
+        }
+        if (!version.isAtLeast(USAGE_ARTIFACTORY_MIN_VERSION)) {
+            String message = String.format("Expected Artifactory version %s or above for usage report, got %s.", USAGE_ARTIFACTORY_MIN_VERSION.toString(), version.toString());
+            throw new VersionException(message, VersionCompatibilityType.INCOMPATIBLE);
+        }
+
+        // Build Artifactory URL.
+        String fullItemUrl = artifactoryUrl + USAGE_API;
+        String encodedUrl = ArtifactoryHttpClient.encodeUrl(fullItemUrl);
+
+        // Create request.
+        String requestBody = new JsonSerializer<UsageReporter>().toJSON(usageReporter);
+        StringEntity entity = new StringEntity(requestBody, "UTF-8");
+        entity.setContentType("application/json");
+        HttpPost request = new HttpPost(encodedUrl);
+        request.setEntity(entity);
+
+        // Send request
+        int connectionRetries = httpClient.getConnectionRetries();
+        HttpResponse httpResponse = null;
+        try {
+            httpResponse = httpClient.getHttpClient().execute(request);
+            StatusLine statusLine = httpResponse.getStatusLine();
+            if (statusLine.getStatusCode() != HttpStatus.SC_OK) {
+                throw new IOException(String.format("Artifactory response: %s %s", statusLine.getStatusCode(), statusLine.getReasonPhrase()));
+            }
+        } finally {
+            // We are using the same client for multiple operations therefore we need to restore the connectionRetries configuration.
+            httpClient.setConnectionRetries(connectionRetries);
```

Comment:
Looks like there's no reason to "restore" the value of *connectionRetries* value, because it never changes after it is initialized in the PreemptiveHttpClient constructor.
I suggest removing the *setConnectionRetries* method, to make sure no one uses it.

Reply:
Since this method is used by the ArtifactoryBaseClient, I didn't remove it completely.
I did remove all unnecessary usages of it.